src/spaces/continuous.jl

```diff
@@ -410,3 +410,230 @@ function Base.iterate(iter::PairIterator, i = 1)
+- `agents`: an iterator of the interacting agents from `interacting_pairs` at each time step.
+"""
+mutable struct FMP_Parameters
+    rho::Float64
+    rho_obstacle::Float64
+    step_inc::Int64
+    terminal_max_dis::Float64
+    c1::Float64
+    c2::Float64
+    vmax::Float64
+    d::Float64
+    r::Float64
+    obstacle_list::Array{Int64}
+    interaction_array::Array{Bool}
+    agents::Dict{Int64, Any}
+    FMP_Parameters() = new()
+end
+
+"""
+    fmp_update_interacting_pairs(model)
+Updates `FMP_Parameters.interaction_array` with the current array of interacting agents by calling
+`interacting_pairs`. It does this once per model step to reduce the potential overhead of calling
+`interacting_pairs` each time `update_vel` is called. 
+"""
+function fmp_update_interacting_pairs(
+    model::ABM{<:ContinuousSpace},
+    )
+
+    # get list of interacting_pairs within some radius
+    agent_iter = interacting_pairs(model, model.FMP_params.r, :all)
+    for agent_id in keys(model.agents)
+        Ni = Int64[]
+        Gi = Int64[]
+        for (i,j) in agent_iter.pairs
+            # handle agent-agent interactions
+            if i == agent_id && model.agents[j].type == :A
+                append!(Ni, j)
+            elseif j == agent_id && model.agents[i].type == :A
+                append!(Ni, i)
+            
+            # handle agent-goal interactions
+            elseif i == agent_id && model.agents[j].type == :T
+                append!(Gi, j)
+            elseif j == agent_id && model.agents[i].type == :T
+                append!(Gi, i)
+            end
+        end
+
+        model.agents[agent_id].Ni = Ni
+        model.agents[agent_id].Gi = Gi
+    end
+
+end
+
+"""
+    fmp_parameter_init()
+A convenience function for initializing the FMP_Parameters struct with typical FMP
+parameters. Users can modify the FMP algorithm parameters through the usage of 
+keyword arguments. 
+"""
+function fmp_parameter_init(;
+    rho = 7.5e6,
+    rho_obstacle = 7.5e6,
+    step_inc = 2,
+    terminal_max_dis = 0.01,
+    c1 = 10,
+    c2 = 10,
+    vmax = 0.1,
+    d = 0.02,
+    obstacle_list=[],
+) where {D,P,T,M}
+
+    r = (3*vmax^2/(2*rho))^(1/3)+d
+    FMP_params = FMP_Parameters()
+    FMP_params.rho = rho
+    FMP_params.rho_obstacle = rho_obstacle
+    FMP_params.step_inc = step_inc
+    FMP_params.terminal_max_dis = terminal_max_dis
+    FMP_params.c1 = c1
+    FMP_params.c2 = c2
+    FMP_params.vmax = vmax
+    FMP_params.d = d
+    FMP_params.r = r
+    FMP_params.obstacle_list = obstacle_list
+    return FMP_params
+end
+
+
+"""
+    fmp_update_vel(model, FMP_params)
+Implements the [Force Based Motion Planning (FMP) algorithm](https://arxiv.org/abs/1909.05415)
+to handle interagent collisions. This function is called each time `agent_step!` is called. This
+function is used in conjunction with the `update_vel!` keyword argument when a `ContinuousSpace`
+is initialized.
```

Comment:
The way this is stated is missleading. This function is not called every time `agent_step!` is called. Rather, the user must manually call it every time they use `agent_step!`, for the algorithm to work.